addons/mail/models/mail_thread.py

```diff
@@ -539,20 +577,14 @@ def message_change_thread(self, new_thread):
     # Automatic log / Tracking
     # ------------------------------------------------------
 
-    @api.model
+    @tools.ormcache()
```

Comment:
are you sure everything gets invalidated properly when modifying custom fields; e.g. writing on `tracking` in `ir.model.fields`?

Reply:
Yes, there is a call to `clear_caches` when modifying a custom field.